src/v/archival/tests/archiver_operations_impl_gtest.cc

```diff
@@ -0,0 +1,1275 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#include "archival/archiver_operations_api.h"
+#include "archival/archiver_operations_impl.h"
+#include "archival/async_data_uploader.h"
+#include "archival/logger.h"
+#include "archival/types.h"
+#include "base/vlog.h"
+#include "bytes/bytes.h"
+#include "bytes/iostream.h"
+#include "cloud_storage/base_manifest.h"
+#include "cloud_storage/partition_manifest.h"
+#include "cloud_storage/remote.h"
+#include "cloud_storage/remote_segment.h"
+#include "cloud_storage/remote_segment_index.h"
+#include "cloud_storage/tx_range_manifest.h"
+#include "cloud_storage/types.h"
+#include "cloud_storage_clients/types.h"
+#include "config/configuration.h"
+#include "consensus.h"
+#include "container/fragmented_vector.h"
+#include "gmock/gmock.h"
+#include "model/fundamental.h"
+#include "model/metadata.h"
+#include "model/record.h"
+#include "model/tests/random_batch.h"
+#include "model/timestamp.h"
+#include "random/generators.h"
+#include "storage/record_batch_utils.h"
+#include "test_utils/scoped_config.h"
+#include "test_utils/test.h"
+#include "utils/available_promise.h"
+#include "utils/retry_chain_node.h"
+
+#include <seastar/core/abort_source.hh>
+#include <seastar/core/future.hh>
+#include <seastar/core/gate.hh>
+#include <seastar/core/iostream.hh>
+#include <seastar/core/lowres_clock.hh>
+#include <seastar/core/shared_ptr.hh>
+#include <seastar/util/log.hh>
+
+#include <gmock/gmock.h>
+
+#include <exception>
+#include <memory>
+#include <stdexcept>
+#include <system_error>
+
+inline ss::logger test_log("arch_op_impl_test");
+
+namespace {
+static cloud_storage::remote_path_provider null_path_provider(std::nullopt);
```

Comment:
this doesn't need to be static?

Reply:
Yes, I'm usually using only 'static' to make sure that the symbol has internal linkage but linter in vscode complains about this.